### PR TITLE
pybind/rbd: parse access and modify timestamps in UTC

### PR DESCRIPTION
**Pull request by idryomov**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/pybind/rbd/rbd.pyx [hunk 1]
```diff
@@ -4094,7 +4094,7 @@ written." % (self.name, ret, length))
             ret = rbd_get_access_timestamp(self.image, &timestamp)
         if ret != 0:
             raise make_ex(ret, 'error getting access timestamp for image: %s' % (self.name))
-        return datetime.fromtimestamp(timestamp.tv_sec)
+        return datetime.utcfromtimestamp(timestamp.tv_sec)
```
Comment (ajarr) [inline src/pybind/rbd/rbd.pyx:4097]: `datetime.fromtimestamp(timestamp.tv_sec, timezone.utc)` is preferred. `utcfromtimestamp` is deprecated from `3.12`
See https://docs.python.org/3/library/datetime.html#datetime.datetime.utcfromtimestamp.

We may want to replace all other `utcfromtimestamp()` calls in the library

**Review by ajarr — CHANGES_REQUESTED** [review]

File: src/pybind/rbd/rbd.pyx (hunk 1 above)
Comment (idryomov) [reply src/pybind/rbd/rbd.pyx:4097]: The result produced by these calls is not identical.  The former is a `datetime` object with with `tzinfo=None`:
```
>>> datetime.datetime.utcfromtimestamp(123456)
datetime.datetime(1970, 1, 2, 10, 17, 36)
>>> datetime.datetime.fromtimestamp(123456, datetime.timezone.utc)
datetime.datetime(1970, 1, 2, 10, 17, 36, tzinfo=datetime.timezone.utc)
```
Given that this is intended to be backported, I'd rather align to the existing usage.  The replacement would need to concern all of `src/pybind/rbd/rbd.pyx` and be main-only.

Comment (ajarr) [reply src/pybind/rbd/rbd.pyx:4097]: Ah okay!

**Review by ajarr — APPROVED** [review]

**Comment by idryomov** [comment]
No failures (`rbd/nvmeof` job isn't expected to work):

https://pulpito.ceph.com/dis-2024-06-10_19:09:24-rbd-wip-dis-testing-distro-default-smithi/
https://pulpito.ceph.com/dis-2024-06-11_07:52:38-rbd-wip-dis-testing-distro-default-smithi/
https://pulpito.ceph.com/dis-2024-06-11_10:44:00-rbd-wip-dis-testing-distro-default-smithi/

**Pull request merged**